utils/streams.js

```diff
@@ -0,0 +1,171 @@
+    })
+    .help('h')
+    .argv;
+  
+  if (argv.action) {
+    if(argv.action === 'cssBundler') actions[argv.action](argv.path);
+    else actions[argv.action](argv.file);
+  }
+}
+
+/* stream helpers*/
+function inputOutput(filePath) { /* 4, babel-node ./utils/streams.js -a io -f ./data/test.txt */
+  if (fs.existsSync(filePath)) fs.createReadStream(filePath).pipe(process.stdout);
+  else console.log('Please specify correct file path');
+}
+
+function transform() { /* 5, node ./utils/streams.js -a transform */
+  stdInInformUser();
+  
+  let write = function (buffer, encoding, next) {
+    this.push(`Transformed to: ${buffer.toString().toUpperCase()}`);
+    next();
+  };
+  
+  process.stdin
+    .pipe(through(write, end))
+    .pipe(process.stdout);
+}
+
+function csvToJsonStream(filePath) { /* 6, babel-node ./utils/streams.js -a csvToJsonStream -f ./data/test.csv */
+  if (fs.existsSync(filePath) && path.extname(filePath) === '.csv') {
+    const readableStream = fs.createReadStream(filePath);
+    readableStream.setEncoding('utf8');
+    readableStream
+      .pipe(through(csvToJsonStreamTransform))
+      .pipe(process.stdout);
+  }
+  else console.log('Please specify correct file path');
+}
+
+function csvToJsonFile(filePath) { /* 7, babel-node ./utils/streams.js -a csvToJsonFile -f ./data/test.csv */
+  if (fs.existsSync(filePath) && path.extname(filePath) === '.csv') {
+    const readableStream = fs.createReadStream(filePath);
+    readableStream.setEncoding('utf8');
+    const fileName = createFilePath(filePath);
+    const writableStream = fs.createWriteStream(fileName);
+    readableStream
+      .pipe(through(csvToJsonStreamTransform, () => {
+        console.log(`${fileName} was generated`);
+      }))
+      .pipe(writableStream);
+  }
+  else console.log('Please specify correct file path');
+}
+
+function stdInInformUser() {
+  process.stdout.write(`Please start typing to transform ...
+  to end input use CTRL + D \n`);
+  process.stdin.setEncoding('utf8');
+}
+
+function csvToJsonStreamTransform(buffer, encoding, next) {
+  let options = {
+    delimiter : ',',
+    quote     : '"'
+  };
+  this.push(JSON.stringify(csvjson.toObject(buffer.toString(), options)));
+  next();
+}
+
+function end() {
+  process.stdout.write('Transform was completed\n');
+}
+
+function createFilePath(filePath, extension = '.json', name) {
+  const pathJsonFile = path.parse(filePath);
+  
+  pathJsonFile.ext = extension;
+  pathJsonFile.base = `${pathJsonFile.name || name}${pathJsonFile.ext}`;
+  return path.format(pathJsonFile);
+}
+
+/* CSS Bundler */
+function cssBundler(path, resultFileName = 'bundle.css') { /* 8, babel-node -- ./utils/streams.js -a cssBundler -p ./data/css */
+  if (fs.existsSync(path)) {
+    const folderPath = getFolderPath(path);
+    const writerableStream = fs.createWriteStream(`${getFolderPath(folderPath)}${resultFileName}`);
+    combineCSSFiles(folderPath, resultFileName, writerableStream);
```

Comment:
There is no guarantee that css from web will be written after css files from a local folder. And it's even worse. There is a chance that you'll write some chunk from local css file, then chunk from remote css and so on.

Reply:
fixed, thx